### PR TITLE
C268

### PR DESCRIPTION
**Pull request by camuffo**
@randomorder 
please tell me if the workaround in entrypoint.sh is acceptable. It's very ugly, but I don't see less invasive alternatives.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,16 @@
 #!/usr/bin/env bash
 geoserver-plugin-download.sh ${CATALINA_BASE}/webapps/${APP_LOCATION}/WEB-INF/lib $PLUGIN_DYNAMIC_URLS
 set -m
+
+# FIXME: this workaround (for a deploy in a Swarm) keeps GS from starting and
+# crashing, because of JDBC not being able to connect to Postgres
+db_err=6
```
Comment (randomorder) [inline entrypoint.sh:7]:
```suggestion
db_err=1
```

File: entrypoint.sh [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+while [ $db_err -eq 6 ]; do
```
Comment (randomorder) [inline entrypoint.sh:8]:
```suggestion
while [ $db_err -ne 0 ]; do
  echo "Checking database availability"
```

File: entrypoint.sh [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  curl -sS "$DATABASE_HOST"
```
Comment (randomorder) [inline entrypoint.sh:9]: Does this work with TCP port 5432?

**Review by randomorder — COMMENTED** [review]
Check my comments. Please test the solution

File: entrypoint.sh (hunk 3 above)
Comment (randomorder) [reply entrypoint.sh:9]: also consider using this script: https://github.com/geosolutions-it/docker-geoserver/blob/master/docker-compose.yml#L11

Comment (camuffo) [reply entrypoint.sh:9]: No. Here I'm only testing the hostname resolution. curl exits with 6 in case of DNS resolution failure.

**Comment by camuffo** [comment]
@randomorder
I tried with pg_isready and it works nicely too. thanks!
Updated the PR.

**Review by randomorder — APPROVED** [review]

**Pull request merged**